### ISSUE TITLE
Amazon Linux 2 - Docker network issue

### ISSUE THREAD

**Issue by JRemitz** [kubernetes/kops#6571]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
`Version 1.11.0 (git-2c2042465)`

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
`GitVersion:"v1.13.2"`

**3. What cloud provider are you using?**
`AWS`

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
Add/update nodes with Amazon Linux 2.  Grab the latest based on kops docs (`us-west-2` example): 
```sh
aws ec2 describe-images --region=us-west-2 --owner=137112412989 \
    --filters "Name=name,Values=amzn2-ami-hvm-2*-gp2" \
    --query 'sort_by(Images,&CreationDate)[-1].{ID:ImageId,name:Name}'
``` 
Based on above - update cluster nodes to use image: `amazon.com/amzn2-ami-hvm-2.0.20190228-x86_64-gp2`

**5. What happened after the commands executed?**
Cluster updates with appropriate nodes. Cluster runs smooth with any deployments for Linux 2 as expected.  Unexpected behavior is with docker daemon's access to external network/internet for non-kubernetes container deployments. It is not allowed either via firewall/or routing issue.  But again, _Kubernetes_ seemingly runs fine.  I'll elaborate below.

**6. What did you expect to happen?**
We build docker containers in Kubernetes, and add the `/var/run/docker.sock` as a mounted volume within our pods/containers for building so we can access the host daemon to start/stop new containers from the pod.  These containers that then start _new_ docker containers without access to internet to update packages, install software, etc - so we cannot build Docker containers from these Amazon Linux 2 nodes.  This can be recreated directly on the host/node without going through kubernetes' pods.  Again, kubernetes pod/container has access just fine, presumably due to CNI configuration?

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**
N/A. I'm not sure it's needed here, but upon request I can.

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**
N/A

**9. Anything else do we need to know?**
Test case once you're running kops with an Amazon Linux 2 node.  After ssh'ing onto the Linux 2 node:
```sh
sudo docker run --rm -it alpine:latest ping 8.8.8.8
```
The above command will hang.  This is unexpected behavior and inconsistent with Debian default (kope.io) and Debian Stretch latest.

However, if you override to host network, it will work. For example:
```sh
sudo docker run --net=host --rm -it alpine:latest ping 8.8.8.8
```
The above command works using the host network.  At first glance, I have only noticed that iptables are missing some default DOCKER chains, however the Linux 2 default FORWARD policy is 'ACCEPT` versus Debian's `DROP`.

Below is a snippet from Debian's `iptables --list` that do NOT exist in Amazon Linux 2 (not sure if related or a red herring):
```
Chain FORWARD (policy DROP)
target     prot opt source               destination
...
DOCKER-ISOLATION  all  --  anywhere             anywhere
DOCKER     all  --  anywhere             anywhere
Chain DOCKER (1 references)                           
target     prot opt source               destination  
.....[other ACCEPT policies]...
                                                      
Chain DOCKER-ISOLATION (1 references)                 
target     prot opt source               destination  
RETURN     all  --  anywhere             anywhere
```

**Bottom line:** besides this being a show-stopper for us to use these Amazon Linux 2 nodes/masters, I don't know if there's a bigger issue that may tie into cluster behavior running these images. I have not moved the masters to Linux 2 but I can't help but wonder if this might tie into some of the root cause for #6341 ?  This behavior is inconsistent for other images like Debian-based which have no issues running the test commands above.  I have not tried with CentOS but that might be another logical comparison to assist in identifying the RCA and whether this is an issue or expected behavior that requires additional settings upon node setup.  Thanks in advance!

**Comment by ElectricWarr** [comment]
@JRemitz Did you ever solve this? I've just run into what sounds like your exact issue, and yours is essentially the only search hit!

**Comment by JRemitz** [comment]
I'm sorry to say that i have not resolved it. Honestly, i gave up trying. Seems like we're the only people attempting this. Works fine with ubuntu or debian images. Have you tried any additional troubleshooting? Any other observations?

**Comment by amoskyler** [comment]
@JRemitz we have the exact same use case as you.

I am running into this issue as well. After a bit of research, it appears it was introduced after a security patch back in February. The root failure is caused because the docker0 bridge has been disabled. I haven't spent quite enough time to figure out a permanent work around.

I've tested using `docker build --network=host .` and confirm it works, at least for my use case.

https://github.com/awslabs/amazon-eks-ami/issues/183

**Comment by v0lkan** [comment]
Not sure if it will help, but overriding `/etc/docker/daemon.json` to use a different DNS helped too.

ref: https://gist.github.com/v0lkan/055328c9cd0a5836ca4f2c53c1e03dca

I was too lazy to figure out the correct iptables rules :)
